lib/Dialect/FIRRTL/InstanceGraph.cpp

```diff
@@ -110,6 +116,28 @@ void InstanceGraph::replaceInstance(InstanceOp inst, InstanceOp newInst) {
   (*it)->instance = newInst;
 }
 
+bool InstanceGraph::isAncestor(FModuleLike child, FModuleOp parent) {
+  DenseSet<InstanceGraphNode *> seen;
+  SmallVector<InstanceGraphNode *> worklist;
+  auto *cn = lookup(child);
+  worklist.push_back(cn);
+  seen.insert(cn);
+  while (!worklist.empty()) {
+    auto *node = worklist.back();
+    worklist.pop_back();
+    if (node->getModule() == parent)
+      return true;
+    for (auto *use : node->uses()) {
+      auto *mod = use->getParent();
+      if (!seen.count(mod)) {
+        seen.insert(mod);
+        worklist.push_back(mod);
+      }
+    }
+  }
+  return false;
+}
```

Comment:
This was clearly a missing API... 👍

Reply:
You might be able to implement it in terms of [inverse_depth_first](https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/ADT/DepthFirstIterator.h#L278) with [this](https://github.com/llvm/circt/blob/main/include/circt/Dialect/FIRRTL/InstanceGraph.h#L283).

Reply:
I was avoiding assuming that the instance graph is a tree.  In theory, at least HW can have (mutually) recursive instantiation.  With parameters, we might have this in firrtl too.